cmd/common/registry.go

```diff
@@ -550,6 +550,9 @@ func addFlags(cmd *cobra.Command, params *params.Params, skipParams []params.Val
 		}
 
 		flag := cmd.PersistentFlags().VarPF(p, p.Key, p.Alias, desc)
+		if p.IsMandatory {
```

Comment:
I saw that you changed a couple of mandatory flags, because they had default values - still, I think we're also using `IsMandatory` to check for non-empty values. Maybe this is cleaner (can have a default value but still can't be set to empty value):

```suggestion
		if p.IsMandatory && p.String() != "" {
```

But I'm fine either way.

Reply:
The purpose of this check in L553 and `MarkPersistentFlagRequired()` is to make the CLI flag required if the parameter is mandatory. 

```bash 
$ ./kubectl-gadget script
Error: required flag(s) "program" not set
```

The check to be sure a mandatory parameter is not empty is done in https://github.com/inspektor-gadget/inspektor-gadget/blob/9dd4b9248ab171266879c030335cab4d7e27d895/pkg/params/params.go#L112-L114

```bash
$ ./kubectl-gadget script --program ""
Error: invalid argument "" for "-e, --program" flag: expected value for "program"
```

So I think we're all good.